pennylane/operation.py

```diff
@@ -762,8 +762,8 @@ def compute_sparse_matrix(*params, **hyperparams):  # pylint:disable=unused-argu
         .. seealso:: :meth:`~.Operator.sparse_matrix`
 
         Args:
-            params (list): trainable parameters of the operator, as stored in the ``parameters`` attribute
-            hyperparams (dict): non-trainable hyperparameters of the operator, as stored in the ``hyperparameters``
+            *params (list): trainable parameters of the operator, as stored in the ``parameters`` attribute
+            **hyperparams (dict): non-trainable hyperparameters of the operator, as stored in the ``hyperparameters``
```

Comment:
Actually is this the standard way to denote variable length arguments? This is the first time I have ever seen it denoted this way in the docs? Might be worth asking @albi3ro if this is the standard approach we want to take from now on?

Reply:
I've never actually looked for an official standard, but the format chosen in PennyLane matches the one introduced by Google (see [this post](https://stackoverflow.com/questions/3898572/what-are-the-most-common-python-docstring-formats) for an overview of formats).
According to the [Google style guide](https://google.github.io/styleguide/pyguide.html) variable-length arguments and generic keyword arguments should be denoted as `*args` and `**kwargs`, respectively.
One could also choose a different convention, but some form of distinction from fixed-length arguments is more user friendly, I'd say.

Reply:
<img width="853" alt="Screenshot 2023-06-13 at 3 50 58 PM" src="https://github.com/PennyLaneAI/pennylane/assets/6364575/0b867587-953b-4dbf-a148-e4a460a26064">

Looks like the style guide says we should be doing it this way. I actually think it much looks much clearer this way. There's probably a lot of places in pennylane

Reply:
I actually really like this way of denoting variable length arguments and keyword arguments. We probably have a bunch of places in our documentation that need to be altered.

We should probably start by letting more people know this is the way we should actually do things.

Reply:
I regex-matched functions that have variable-length argument lists across PennyLane and replaced them. But I probably missed some.